homeassistant/components/zwave_js/logbook.py

```diff
@@ -0,0 +1,139 @@
+        """Describe Z-Wave JS notification event."""
+        device = dev_reg.devices[event.data[ATTR_DEVICE_ID]]
+        # Z-Wave JS devices always have a name
+        device_name = device.name_by_user or device.name
+        assert device_name
+
+        command_class = event.data[ATTR_COMMAND_CLASS]
+        command_class_name = event.data[ATTR_COMMAND_CLASS_NAME]
+
+        data: dict[str, str] = {LOGBOOK_ENTRY_NAME: device_name}
+        prefix = f"fired {command_class_name} CC 'notification' event"
+
+        if command_class == CommandClass.NOTIFICATION:
+            label = event.data[ATTR_LABEL]
+            event_label = event.data[ATTR_EVENT_LABEL]
+            return {
+                **data,
+                LOGBOOK_ENTRY_MESSAGE: f"{prefix} '{label}': '{event_label}'",
+            }
+
+        if command_class == CommandClass.ENTRY_CONTROL:
+            event_type = event.data[ATTR_EVENT_TYPE]
+            data_type = event.data[ATTR_DATA_TYPE]
+            return {
+                **data,
+                LOGBOOK_ENTRY_MESSAGE: (
+                    f"{prefix} for event type '{event_type}' with data type '{data_type}'"
+                ),
+            }
+
+        if command_class == CommandClass.SWITCH_MULTILEVEL:
+            event_type = event.data[ATTR_EVENT_TYPE]
+            direction = event.data[ATTR_DIRECTION]
+            return {
+                **data,
+                LOGBOOK_ENTRY_MESSAGE: (
+                    f"{prefix} for event type '{event_type}': '{direction}'"
+                ),
+            }
+
+        return {**data, LOGBOOK_ENTRY_MESSAGE: prefix}
+
+    @callback
+    def async_describe_zwave_js_value_notification_event(
+        event: Event,
+    ) -> dict[str, str]:
+        """Describe Z-Wave JS value notification event."""
+        device = dev_reg.devices[event.data[ATTR_DEVICE_ID]]
+        # Z-Wave JS devices always have a name
+        device_name = device.name_by_user or device.name
+        assert device_name
+
+        command_class = event.data[ATTR_COMMAND_CLASS_NAME]
+        label = event.data[ATTR_LABEL]
+        value = event.data[ATTR_VALUE]
+
+        return {
+            LOGBOOK_ENTRY_NAME: device_name,
+            LOGBOOK_ENTRY_MESSAGE: f"fired {command_class} CC 'value notification' event for '{label}': '{value}'",
+        }
+
+    @callback
+    def async_describe_zwave_js_value_updated_event(
+        event: Event,
+    ) -> dict[str, str]:
+        """Describe Z-Wave JS value updated event."""
+        device = dev_reg.devices[event.data[ATTR_DEVICE_ID]]
+        # Z-Wave JS devices always have a name
+        device_name = device.name_by_user or device.name
+        assert device_name
+
+        entity_id = event.data[ATTR_ENTITY_ID]
+        value = event.data[ATTR_VALUE]
+
+        return {
+            LOGBOOK_ENTRY_NAME: device_name,
+            LOGBOOK_ENTRY_ENTITY_ID: entity_id,
+            LOGBOOK_ENTRY_MESSAGE: f"fired 'value updated' event: '{value}'",
```

Comment:
We should only include things in the logbook that are relevant to the user and that are not already represented otherwise. A value updated event would already be in the logbook because the entity that represents that value changed state?

Reply:
thanks for mentioning that, I actually forgot about this use case so you're right that this needs a change, but for a different reason. The `value updated` event is only fired for devices that have exhibited odd behavior. There was one device in question that triggered the addition of this event where all value updates were reported on a single endpoint, including for values that were on other endpoints. So this is still useful to show in the logbook, but only when it's a value update for an entity other than the entity that is in the event.

Reply:
actually I'm just going to remove this for now - I have to think through whether this is really valuable as a logbook entry, and if so how/when